system/jlib/jlog.cpp

```diff
@@ -292,28 +293,33 @@ void LogMsgJobInfo::deserialize(MemoryBuffer & in)
 }
 
 static LogMsgJobInfo globalDefaultJobInfo(UnknownJob, UnknownUser);
+static LogMsgTraceInfoId globalDefaultTraceInfoId = UnknownTraceInfoId;
 
 // NOTE - extern thread_local variables are very inefficient - don't be tempted to expose the variables directly
 
 static  TraceFlags defaultTraceFlags = TraceFlags::Standard;
 static thread_local LogMsgJobInfo defaultJobInfo;
+static thread_local LogMsgTraceInfoId defaultTraceInfoId = UnknownTraceInfoId;
 static thread_local TraceFlags threadTraceFlags = TraceFlags::Standard;
 static thread_local const IContextLogger *default_thread_logctx = nullptr;
 
 const LogMsgJobInfo unknownJob(UnknownJob, UnknownUser);
+const LogMsgTraceInfo unknownTrace(UnknownTraceInfoId);
 
-void getThreadLoggingInfo(const IContextLogger * &_logctx, TraceFlags &_traceFlags)
+void getThreadLoggingInfo(const IContextLogger * &_logctx, TraceFlags &_traceFlags, unsigned __int64 &_traceInfoId)
 {
     _logctx = default_thread_logctx;
     _traceFlags = threadTraceFlags;
+    _traceInfoId = defaultTraceInfoId;
 }
 
-void resetThreadLogging(const IContextLogger *_logctx, TraceFlags _traceFlags)
+void resetThreadLogging(const IContextLogger *_logctx, TraceFlags _traceFlags, unsigned __int64 _traceInfoId)
 {
     // Note - as implemented the thread default job info is determined by what the global one was when the thread was created.
     // There is an alternative interpretation, that an unset thread-local one should default to whatever the global one is at the time the thread one is used.
     // In practice I doubt there's a lot of difference as global one is likely to be set once at program startup
     defaultJobInfo = globalDefaultJobInfo;
+    setDefaultTraceInfo(_traceInfoId, true);
```

Comment:
probably easier to undestand setting
```
 defaultTraceInfoId = _traceInfoId;
```
Since this is an internal restoring operation.